Automattic/wp-calypso
# Add Domain Connect components for G Suite and Zoho Mail.

This PR adds support for setting up the DNS records for Zoho Mail and G Suite email services from the _Name Servers and DNS_ section of the Domain Settings page for a given domain associated with a site by using Domain Connect templates.

This feature is designed for a user who has a WordPress.com site; has purchased G Suite, Office 365, or Zoho Mail services elsewhere; and who now needs to set up a domain that uses WordPress.com name servers to work with their email service.

When the user selects the email service they want to set up, they will be asked to enter a verification token required by the service. The token string format for each service is different, but the first part of the token string is shown in the placeholder text to help the user understand what is being asked for.

Once the user has entered a valid token for the selected service, the "Set up..." button is activated and when clicked, the DNS records required to use the email service will be created for the domain.

This PR builds on previous work done to add Office365 DNS records and specifically makes the following additions and change:
- Created GSuite and ZohoMail Components based on the Office365 Component
- Changed the className of the jsx forms to `dns__template-form` since it it more generic than `dns__office365`

NOTE: This PR is dependent on D5465-code which adds the actual Domain Connect templates that are used to apply the DNS records (Committed 2017-05-05 20:10 UTC).

Steps to test this PR:

1. Run the `add/new-domain-connect-templates` branch of Calypso.
2. Point `public-api.wordpress.com` and `wordpress.com` to your test environment
3. Navigate to My Sites ->Domains, Select a domain, Select Name Servers and DNS
4. Select G Suite, Office365, or Zoho Mail
5. Fill out the form with the appropriate token (See below for example tokens.)
6. Clicking on the "Set up G Suite", "Set up Office 365", or "Set up Zoho Mail" button to apply the DNS records. (Clicking the button multiple times simply re-applies the same records and has no negative effects.)
7. View the DNS records page to confirm that the records specified in the templates were applied correctly.

![2017-05-16 17 30 11](https://cloud.githubusercontent.com/assets/1379730/26129303/b1e060be-3a5d-11e7-97c4-d79025e3b077.gif)

Starting over and applying a template for a different email service will remove all the previous MX records, but will not remove any other `TXT`, `CNAME`, or `SRV` records.

Example verification token for G Suite:
`google-site-verification=1234567890abcdefghijklmnopqrstuvwxyzABCDEFG`
("google-site-verification=" followed by 43 alpha numerics)

Example verification token for Office365:
`MS=ms1234`
("MS=ms" followed by 4-20 digits.)

Example verification token for Zoho Mail:
`zb1234asdfASDF`
("zb" followed by an unspecified number of alpha numerics)
